ui/plugin-system/src/model/plugin-base.ts

```diff
@@ -20,7 +20,7 @@ export interface Plugin<Spec> {
   /**
    * React component for editing the plugin's options in the UI.
    */
-  OptionsEditorComponent: React.ComponentType<OptionsEditorProps<Spec>>;
+  OptionsEditorComponent?: React.ComponentType<OptionsEditorProps<Spec>>;
```

Comment:
should this still be part of the base type for all plugin implementations if panel plugin doesn't use this?

Reply:
yeah I'm thinking we can just add `OptionsEditorComponent` to  these types `VariablePlugin` and `TimeSeriesQueryPlugin`  instead having it in the base `Plugin` type since not every plugin will require an options editor component